### PR TITLE
UAST: first impl - UDF using bblfsh/clinet-scala

### PR DESCRIPTION
**Pull request by bzz**
This is initial approach of implementing `.parseUASTs()` using a custom UDF delegating to [bblfsh/client-scala](https://github.com/bblfsh/client-scala)

This implementation is sub-optimal, as it does not re-use bblfsh client instances between rows. Following a better approach from scratch is bit tricky: it consists of `.mapPartitions` over the DF with some schema modification (adding a column), and requires more time.
 
This approach is nevertheless useful, as both implementation exhibit the same API and thus can be wrapped in pySpark asap, and optimized independently.

Includes
 - [x] bblfsh server to CI
 - [x] `.parseUAST()` \w UDF
 - [x] ability to do `./sbt dependencyTree` (required to resolve dependencies conflicts)
 - [x] small bugfix for `Implicits.checkCols`

Missing things:
 - [x] tests
 - [x] parametrize bblfsh server address/port AKA pass non-column arguments to UDF
 - [x] switch to original bblfsh/client-scala, as soon as https://github.com/bblfsh/client-scala/pull/1 is merged
 - [x] ~performance optimization: re-use connections~ Will be explored in sub-sequent PRs.

### CODE REVIEW

**Comment by bzz** [comment]
Rebased on latest master

**Comment by bzz** [comment]
Thank you for the prompt reviews!

**Comment by bzz** [comment]
It's ready to be merged.

File: src/main/scala/tech/sourced/api/Implicits.scala [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
-import org.apache.spark.sql.functions._
 import org.apache.spark.sql.{DataFrame, SparkSession}
-import tech.sourced.api.customudf.{CustomUDF, ClassifyLanguagesUDF}
+import tech.sourced.api.udf.{ExtractUASTsUDF, ClassifyLanguagesUDF, CustomUDF}
 
 object Implicits {
 
   implicit class SessionFunctions(session: SparkSession) {
     def registerUDFs(): Unit = {
+      ExtractUASTsUDF.bblfshHost = session.sparkContext.getConf.get("tech.sourced.bblfsh.grpc.host", "0.0.0.0")
```
Comment (erizocosmico) [inline src/main/scala/tech/sourced/api/Implicits.scala]: I'm gonna add functions to set this in `SparkAPI` in #45, maybe we should wait until it's merged to reuse the key constants

**Comment by erizocosmico** [comment]
@bzz Just stopped the build because it has been stuck for almost 30m

**Comment by bzz** [comment]
Rebased on latest master. But CI takes forever :/

**Comment by bzz** [comment]
Rebased on #48 which should be merged first

**Comment by bzz** [comment]
All done now, CI is green, merging if there is no further discussion

**Pull request merged**